lib/src/carry_save_multiplier.dart

```diff
@@ -0,0 +1,186 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// carry_save_multiplier.dart
+// Implementation of pipeline multiplier module.
+//
+// 2023 February 21
+// Author: Max Korbel <max.korbel@intel.com>
+//
+
+import 'package:rohd/rohd.dart';
+
+/// A datatype class that hold result from [FullAdder].
+class FullAdderResult {
+  /// The output sum from full adder.
+  final sum = Logic(name: 'sum');
+
+  /// The output carry-out from full adder.
+  final cOut = Logic(name: 'c_out');
+}
+
+/// A [FullAdder] module that can perform simple adder. The full adder output
+/// can be defined in the truth table.
+class FullAdder extends Module {
```

Comment:
A full adder is a good component all by itself, we should include it in a separate file?